code/game/machinery/doors/airlock.dm

```diff
@@ -500,6 +520,31 @@ About the new airlock wires panel:
 				sleep(6)
 				update_icon(AIRLOCK_CLOSED)
 
+
+/obj/machinery/door/airlock/proc/change_paintjob(obj/item/airlock_painter/W, mob/user)
+	if(!paintable)
+		to_chat(user, "<span class='warning'>You cannot paint this type of airlock.</span>")
+		return FALSE
+
+	if(airlock_material == "glass" && W.paint_setting == "Freezer") // Freezer doesn't have a glass version
```

Comment:
I think it's better to inspect airlock's assembly .glass variable to see whether it has glass now, and indirect assembly .noglass var to see if it can be installed at all (instead of hardcoding freezer).

Reply:
You can't use the painter on assemblies, only fully constructed airlocks. Originally I wanted to do that but it would be more annoying. Airlocks that have glass will have the var `airlock_material == "glass"`

Reply:
Oh, you're right, the original assembly is hard to get from here, my bad!

You can still get the noglass as `(new src.assemblytype() ).noglass` (again, instead of hardcoding the freezer)

Reply:
Hm, yeah that actually works, neat little trick